flask_project/url_scanner.py

```diff
@@ -31,3 +32,7 @@ def get_domain(url: str) -> str:
     """
     parsed_url = urlparse(url)
     return parsed_url.netloc
+
+def extract_urls(text):
+    url_extract_pattern = "https?:\\/\\/(?:www\\.)?[-a-zA-Z0-9@:%._\\+~#=]{1,256}\\.[a-zA-Z0-9()]{1,6}\\b(?:[-a-zA-Z0-9()@:%_\\+.~#?&\\/=]*)"
```

Comment:
Pattern should be present as constant/class variable/configuration instead of variable that will be reinitialized during each call of the function.